lib/stat_tracker.rb

```diff
@@ -146,4 +139,93 @@ def lowest_scoring_visitor
     end
     team_name_from_id(teams_hash.key(teams_hash.values.min))
   end
-end
+
+  def count_of_teams
+   @teams_reader.length
+  end
+
+  def unique_total_goals
+    goal_totals = []
+      @games_reader.each do |row|
+        if goal_totals.include?(row[:away_goals].to_i + row[:home_goals].to_i) == false
+          goal_totals << row[:away_goals].to_i + row[:home_goals].to_i
+        end
+      end
+    goal_totals
+  end
+
+  def highest_total_score
+    unique_total_goals.max
+  end
+
+  def lowest_total_score
+    unique_total_goals.min
+  end
+
+  def total_number_of_games
+    @games_reader.length
+  end
+
+  def percentage_home_wins
+   home_win_total = @game_teams_reader.count {|row| row[:result] == "WIN" && row[:hoa] == "home"}.to_f.round(2)
+   home_win_total*100/total_number_of_games
+  end
+
+  def percentage_visitor_wins
+    home_visitor_total = @game_teams_reader.count {|row| row[:result] == "WIN" && row[:hoa] == "away"}.to_f.round(2)
+    home_visitor_total*100/total_number_of_games
+  end
+
+  def percentage_ties
+    tie_total = @game_teams_reader.count {|row| row[:result] == "TIE" && row[:hoa] == "home"}.to_f.round(2)
+    tie_total*100/total_number_of_games
+  end
+
+  def team_finder(team_id)
+    @teams_reader.find do |row|
+      row[:team_id] == team_id
+    end
+  end
+
+  def team_info(team_id)
+    Team.new(team_finder(team_id)).team_labels
+  end
+
+  def average_win_percentage(team_id)
+    team_win_total = @game_teams_reader.count {|row| row[:result] == "WIN" && row[:team_id] == team_id}.to_f.round(2)
+    total_team_games = @game_teams_reader.count {|row| row[:team_id] == team_id}
+    team_win_total*100/total_team_games
+  end
+
+  def count_of_games_by_season
+    seasons = {}
```

Comment:
This might be a good spot to use Hash.new(0) so each season can be incremented without using a conditional.

Reply:
The conditional is there for the purpose of creating a new key value pair if it hadn't already existed rather than just incrementing. 

I'm not sure how you're thinking of setting it up exactly, but feel free to make the edit yourself or suggest it here if you don't want to just overwrite and if it accomplishes the same goal in the end but takes less lines, I'm good with it!